server/src/main/java/org/opensearch/common/xcontent/JsonToStringXContentParser.java

```diff
@@ -74,8 +74,7 @@ public XContentParser parseObject() throws IOException {
         return JsonXContent.jsonXContent.createParser(this.xContentRegistry, this.deprecationHandler, String.valueOf(jString));
     }
 
-    private void parseToken(StringBuilder path, String currentFieldName) throws IOException {
-
+    private void parseToken(StringBuilder path, String currentFieldName, boolean popName) throws IOException {
```

Comment:
Thanks for the fix @naomichi-y , do you think we could refactor the method to use just `Stack<String>` to collect the current path instead of `StribgBuffer` + boolean flag manipulation? (to be fair, it is pretty difficult to follow the current implementation and it is becoming even more complicated). Thank you.

Reply:
@reta 
I looked into the code a bit, and it seems that removing popName might be difficult.
The parseToken method parses the JSON recursively and removes trailing hierarchies from the field names depending on the type of the parent token.
However, I haven't been able to read the original code in its entirety, and I'm not very familiar with Java, so I might be misunderstanding something...

https://github.com/opensearch-project/OpenSearch/blob/7f7ee1059b0f0f743abaf85ade14e7ab9b9c02c4/server/src/main/java/org/opensearch/common/xcontent/JsonToStringXContentParser.java#L109